Source/WebKit/GPUProcess/graphics/RemoteGraphicsContextGL.cpp

```diff
@@ -261,6 +261,16 @@ bool RemoteGraphicsContextGL::webXREnabled() const
     return false;
 }
 
+bool RemoteGraphicsContextGL::webXRPromptAccepted() const
+{
+#if ENABLE(WEBXR) && PLATFORM(COCOA) && !PLATFORM(IOS_FAMILY_SIMULATOR)
+    auto currentAcceptedValue = GPUProcess::singleton().immersiveModeProcessIdentity();
```

Comment:
This is not thread safe

Reply:
I don’t think it needs to be because only one message can be sent from the user’s device at a time, right?

Reply:
I think with https://github.com/WebKit/WebKit/pull/31404 the lack of thread safety in this member function is a non-issue, since `webXRPromptAccepted` is only called during immersive mode or shortly after the immersion session has ended. It would not be possible from the UI layer to reproduce a race here in that window.

Reply:
It's definitively possible to se torn read. GL reads this in GL work queue, the message handler for the UI process -sent message write this in main thread.